File: backend/src/main/java/cn/etarch/mao/session/ws/WsEvent.java
package cn.etarch.mao.session.ws;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Map;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class WsEvent {
    private String type;
    private Long sessionId;
    private Map<String, Object> data;

    public static WsEvent of(String type, Long sessionId, Map<String, Object> data) {
        return new WsEvent(type, sessionId, data);
    }
}

File: backend-ts/src/session/ws/ws-event.ts
export interface WsEvent {
  type: string;
  sessionId: number | null;
  data: Record<string, unknown> | null;
}

export function wsEvent(type: string, sessionId: number | null, data: Record<string, unknown>): WsEvent {
  return { type, sessionId, data };
}
